src/commands/gate/evaluate.ts

```diff
@@ -85,33 +88,46 @@ export class GateEvaluateCommand extends Command {
       return 0
     }
 
-    // To be extended with retries, error handling, etc.
-    return api
-      .evaluateGateRules(evaluateRequest, this.context.stdout.write.bind(this.context.stdout))
+    return retryRequest(
+      () => api.evaluateGateRules(evaluateRequest, this.context.stdout.write.bind(this.context.stdout)),
+      {
+        onRetry: (e, attempt) => {
+          this.context.stderr.write(renderEvaluationRetry(attempt, e))
+        },
+        retries: 5,
+      }
+    )
       .then((response) => {
-        return this.handleEvaluationResponse(response.data.data.attributes)
+        return this.handleSuccessfulEvaluation(response.data.data.attributes)
       })
       .catch((error) => {
-        // TODO Handle unavailability etc.
-        this.context.stderr.write(renderGateEvaluationError(error))
-
-        return 1
+        return this.handleEvaluationError(error)
       })
   }
 
-  private handleEvaluationResponse(evaluationResponse: EvaluationResponse): number {
+  private handleSuccessfulEvaluation(evaluationResponse: EvaluationResponse): number {
```

Comment:
do we want to name this `handleEvaluationSuccess`? To follow the convention with `handleEvaluationError`

Reply:
Makes sense, renamed!